### PR TITLE
Add --protocol options for alignment with different data types (minimap2)

### PR DESCRIPTION
**Pull request by lwratten**
Added the `--protocol` and `--stranded` flags.
Changed the `minimap2` alignment calls to work with this data - `graphmap2` (required for RNA) has less documentation so I will read up and submit in a future PR if that's cool?
At the moment I have tested the alignment with `directRNA`, `cDNA` and `stranded cDNA` data successfully.
Will need to update the tests to pass test suite.

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] If necessary, also make a PR on the [nf-core/nanoseq branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/nanoseq)
 - [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/nanoseq/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -203,6 +205,24 @@ Skip basecalling and demultiplexing step with Guppy
 
 ## Alignment
 
+### `--protocol`
+
+Specifies the data type being inputted (DNA, cDNA or directRNA)
```
Comment (drpatelh) [inline docs/usage.md]:
```suggestion
Specifies the type of data that was sequenced i.e. "DNA", "cDNA" or "directRNA".
```

File: main.nf [hunk 2]
```diff
@@ -39,6 +39,8 @@ def helpMessage() {
       --skip_demultiplexing [bool]    Skip basecalling and demultiplexing step with Guppy
 
     Alignment
+      --protocol [str]                Specifies the type of data being inputted (DNA, cDNA or directRNA)
```
Comment (drpatelh) [inline main.nf]:
```suggestion
      --protocol [str]                Specifies the type of data that was sequenced i.e. "DNA", "cDNA" or "directRNA".
```

File: main.nf [hunk 3]
```diff
@@ -39,6 +39,8 @@ def helpMessage() {
...
+      --stranded [bool]               Specifies if the inputted data is stranded (default: false, automatically true when using --protocol directRNA)
```
Comment (drpatelh) [inline main.nf]:
```suggestion
      --stranded [bool]               Specifies if the data is strand-specific. Automatically activated when using --protocol directRNA (default: false)
```

File: docs/usage.md [hunk 4]
```diff
@@ -203,6 +205,24 @@ Skip basecalling and demultiplexing step with Guppy
...
+
+### `--stranded`
+
+Specifies whether the inputted fastq files contain stranded data (default: false)
```
Comment (drpatelh) [inline docs/usage.md]:
```suggestion
Specifies if the data is strand-specific. Automatically activated when using --protocol directRNA (default: false)
```

File: docs/usage.md [hunk 5]
```diff
@@ -203,6 +205,24 @@ Skip basecalling and demultiplexing step with Guppy
...
+
+The specific command line arguments for `graphmap` and `minimap2` using `--protocol` and `--stranded` are specified below:
```
Comment (drpatelh) [inline docs/usage.md]:
```suggestion
When using `--protocol`/`--stranded` the following command-line arguments will be set for `minimap2` and `graphmap`:
```

File: docs/usage.md [hunk 6]
```diff
@@ -203,6 +205,24 @@ Skip basecalling and demultiplexing step with Guppy
...
+
+| `nanoseq` input              | `minimap2` presets  | `graphmap` presets |
+|------------------------------|---------------------|--------------------|
+| `--protocol DNA`             | -ax map-ont         | tba                |
+| `--protocol cDNA`            | -ax splice          | tba                |
+| `--protocol directRNA`       | -ax splice -uf -k14 | tba                |
+| `--protocol cDNA --stranded` | -ax splice -uf      | tba                |
+
+
```
Comment (drpatelh) [inline docs/usage.md]:
```suggestion
```

Comment (drpatelh) [reply docs/usage.md]: Removing this line should get the markdownlint tests passing. Pipeline tests are purring 😃

**Comment by lwratten** [comment]
Doc rephrase suggestions are all committed 😄

**Comment by drpatelh** [comment]
Great! Thanks :+1:

**Pull request merged**